EvidenceApi/V1/Gateways/DocumentsApiGateway.cs

```diff
@@ -71,6 +71,22 @@ public async Task<S3UploadPolicy> CreateUploadPolicy(Guid id)
             return await DeserializeResponse<S3UploadPolicy>(response).ConfigureAwait(true);
         }
 
+        public async Task UploadDocument(Guid documentId, DocumentSubmissionRequest request)
+        {
+            var uri = new Uri($"api/v1/documents/{documentId}", UriKind.Relative);
+            _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(_options.DocumentsApiPostDocumentsToken);
+            MultipartFormDataContent formDataContent = new MultipartFormDataContent();
+            var document = request.Document;
+            formDataContent.Headers.ContentType.MediaType = "multipart/form-data";
+            formDataContent.Headers.ContentDisposition = new ContentDispositionHeaderValue("document");
+            formDataContent.Add(new StreamContent(document.OpenReadStream()), document.Name, document.FileName);
+            var response = await _client.PostAsync(uri, formDataContent).ConfigureAwait(true);
+            if (response.StatusCode != HttpStatusCode.OK)
+            {
+                throw new DocumentsApiException($"Incorrect status code returned: {response.StatusCode}");
```

Comment:
Why would the status code returned be incorrect? Shouldn't this say what was the problem?

Reply:
We're expecting the response to be `OK` but it could be other responses, see the [DocumentsApi controller](https://github.com/LBHackney-IT/documents-api/blob/e0e06f3c554c8910c4e75a80dc19396fb06e5291/DocumentsApi/V1/Controllers/DocumentsController.cs#L45-L60)

I followed how the other methods in this gateway handle HTTP responses, which as you mentioned perhaps isn't ideal because they do not log what the exact issue was. EvidenceAPI is only expecting an object in JSON, and we haven't figured out a good way to pass the exact error back to the caller